internal/executor/instance/container/run.go

```diff
@@ -0,0 +1,329 @@
+		return err
+	}
+
+	logger.Debugf("creating container using working volume %s", params.WorkingVolumeName)
+	input := containerbackend.ContainerCreateInput{
+		Image: params.Image,
+		Entrypoint: []string{
+			params.Platform.ContainerAgentPath(),
+			"-api-endpoint",
+			config.ContainerEndpoint,
+			"-server-token",
+			config.ServerSecret,
+			"-client-token",
+			config.ClientSecret,
+			"-task-id",
+			strconv.FormatInt(config.TaskID, 10),
+			"-command-from",
+			params.CommandFrom,
+			"-command-to",
+			params.CommandTo,
+		},
+		Env: make(map[string]string),
+		Mounts: []containerbackend.ContainerMount{
+			{
+				Type:   containerbackend.MountTypeVolume,
+				Source: params.AgentVolumeName,
+				Target: params.Platform.ContainerAgentVolumeDir(),
+			},
+		},
+		Resources: containerbackend.ContainerResources{
+			NanoCPUs: int64(params.CPU * nano),
+			Memory:   int64(params.Memory * mebi),
+		},
+	}
+
+	if runtime.GOOS == "linux" {
+		if heuristic.GetCloudBuildIP(ctx) != "" {
+			// Attach the container to the Cloud Build network for RPC the server
+			// to be accessible in case we're running in Cloud Build and the CLI
+			// itself is containerized (so we can't mount a Unix domain socket
+			// because we don't know the path to it on the host)
+			input.Network = heuristic.CloudBuildNetworkName
+		}
+
+		// Disable SELinux confinement for this container
+		//
+		// This solves the following problems when SELinux is enabled:
+		// * agent not being able to connect to the CLI's Unix socket
+		// * task container not being able to read project directory files when using dirty mode
+		input.DisableSELinux = true
+	}
+
+	// Mount the  directory with the CLI's Unix domain socket in case it's used,
+	// assuming that we run in the same mount namespace as the Docker daemon
+	if strings.HasPrefix(config.ContainerEndpoint, "unix:") {
+		socketPath := strings.TrimPrefix(config.ContainerEndpoint, "unix:")
+		socketDir := filepath.Dir(socketPath)
+
+		input.Mounts = append(input.Mounts, containerbackend.ContainerMount{
+			Type:   containerbackend.MountTypeBind,
+			Source: socketDir,
+			Target: socketDir,
+		})
+	}
+
+	if config.DirtyMode {
+		// In dirty mode we mount the project directory from host
+		input.Mounts = append(input.Mounts, containerbackend.ContainerMount{
+			Type:   containerbackend.MountTypeBind,
+			Source: config.ProjectDir,
+			Target: params.WorkingDirectory,
+		})
+	} else {
+		// Otherwise we mount the project directory's copy contained in a working volume
+		input.Mounts = append(input.Mounts, containerbackend.ContainerMount{
+			Type:   containerbackend.MountTypeVolume,
+			Source: params.WorkingVolumeName,
+			Target: params.WorkingDirectory,
+		})
+	}
+
+	for _, volume := range params.Volumes {
+		input.Mounts = append(input.Mounts, containerbackend.ContainerMount{
+			Type:     containerbackend.MountTypeBind,
+			Source:   volume.Source,
```

Comment:
Should we check here that the source exists? And create an empty directory if not. Or Docker won't fail on a missing folder 🤔

Reply:
IMO we should just create it. But I'm not sure is Docker already does it. Will be great to modify the test to try to write to some non-existing mounted folder and see what will happen

Reply:
>But I'm not sure is Docker already does it.

It doesn't, you can easily try it out like that: `docker run -it -v /non-existent:/non-existent --rm debian:latest`.